deps/zlib.BUILD.bazel

```diff
@@ -101,17 +93,13 @@ cc_library(
         "-I.",
         "-Wall",
     ] + select({
-        ":mingw_gcc_compiler": [
-            "-fpermissive",
-        ],
         "@platforms//os:windows": [
-            "-fpermissive",
-            "/O2",
-            "/wd4127",  # conditional expression is constant
-            "/wd4242",  # possible loss of data
-            "/wd4244",  # possible loss of data
-            "/wd4711",  # it inlined a function
-            "/wd5045",  # padding bytes inserted
+            "-O2",
+            #"/wd4127",  # conditional expression is constant
+            #"/wd4242",  # possible loss of data
+            #"/wd4244",  # possible loss of data
+            #"/wd4711",  # it inlined a function
+            #"/wd5045",  # padding bytes inserted
```

Comment:
If we no longer need these, we can remove them altogether rather than leaving them commented out. Though knowing why we're changing this now would be nice, too.